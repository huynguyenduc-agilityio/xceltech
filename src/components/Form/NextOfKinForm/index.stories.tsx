import type { Meta, StoryObj } from '@storybook/react';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import NextOfKinForm from '.';

const queryClient = new QueryClient();

const meta = {
  title: 'Components/NextOfKinForm',
  component: NextOfKinForm,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  render: () => (
    <QueryClientProvider client={queryClient}>
      <div className="flex items-center justify-center w-screen py-20 bg-blue-light">
        <div className="bg-white w-[988px] rounded-[15px]">
          <div className="p-10">
            <NextOfKinForm />
          </div>
        </div>
      </div>
    </QueryClientProvider>
  ),
} satisfies Meta<typeof NextOfKinForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
