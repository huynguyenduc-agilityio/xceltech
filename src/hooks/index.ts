export * from './useToast';
export * from './useConfirm';
export * from './useAuth';
export * from './useUser';
export * from './useLeave';
export * from './useAccount';
export * from './usePagination';
export * from './useNotification';
export * from './useJob';
export * from './useDocument';
export * from './useEducation';
export * from './useGuarantor';
export * from './useFamily';
export * from './useFinancial';
