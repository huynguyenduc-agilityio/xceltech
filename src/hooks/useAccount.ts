import { useQuery } from '@tanstack/react-query';

// Constants
import { accountsQueryKeys } from '@/constants';

// Types
import { IInfoUser } from '@/types';

// Services
import { getAccounts } from '@/services';

export const useGetLeaveAccounts = (userId: string) => {
  const { data: leaveAccounts, isFetching: isLeaveAccountsLoading } = useQuery({
    queryKey: [...accountsQueryKeys.list()] as const,
    queryFn: getAccounts,
    select: (data) =>
      data
        ?.filter(({ id }: { id: string }) => id !== userId)
        .map(({ id, firstName, lastName }: IInfoUser) => ({
          value: id,
          label: `${firstName} ${lastName}`,
        })),
  });

  return { leaveAccounts, isLeaveAccountsLoading };
};
