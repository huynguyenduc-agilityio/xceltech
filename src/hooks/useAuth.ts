import { useNavigate, useLocation } from 'react-router-dom';
import { useMutation } from '@tanstack/react-query';

//Types
import { IActiveUser } from '@/types';

// Constants
import {
  ADMIN_PAGE,
  AUTHENTICATION_PAGE,
  RoleAuthentication,
  USER_PAGE,
} from '@/constants';

// Services
import { activateAccount, loginUser, registerUser } from '@/services';

// Stores
import { useUserActions } from '@/stores';

export const useLogin = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const { setUser } = useUserActions();

  const {
    mutateAsync: handleLoginUser,
    isPending: isLoginLoading,
    isError: isLoginError,
  } = useMutation({
    mutationFn: loginUser,
    onSuccess: (data) => {
      // Save data in local storage
      if (data) {
        const path = location.pathname;
        const role = data.user.role;
        const isAdmin =
          role === RoleAuthentication.Admin &&
          path === AUTHENTICATION_PAGE.ADMIN_SIGN_IN;

        const isUser =
          [RoleAuthentication.Employee, RoleAuthentication.Candidate].includes(
            role as RoleAuthentication,
          ) && path === AUTHENTICATION_PAGE.USER_SIGN_IN;

        if (isAdmin) {
          setUser(data);
          navigate(ADMIN_PAGE.DASHBOARD);
        }

        if (isUser) {
          setUser(data);
          navigate(USER_PAGE.DASHBOARD);
        }
      }
    },
  });

  return { isLoginLoading, isLoginError, handleLoginUser };
};

export const useRegister = () => {
  const { mutateAsync: handleRegisterUser, isPending: isRegisterLoading } =
    useMutation({
      mutationFn: registerUser,
    });

  return {
    handleRegisterUser,
    isRegisterLoading,
  };
};

export const useActivateAccount = ({ uidb64, token }: IActiveUser) => {
  const { mutateAsync: handleActivateUser, isPending: isActivateLoading } =
    useMutation({
      mutationFn: () => activateAccount({ uidb64, token }),
    });

  return {
    handleActivateUser,
    isActivateLoading,
  };
};
