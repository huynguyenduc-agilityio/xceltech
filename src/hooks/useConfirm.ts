import { useConfirmDialogStore } from '@/stores';

export const useConfirm = () => {
  return useConfirmDialogStore((state) => state.confirm);
};
