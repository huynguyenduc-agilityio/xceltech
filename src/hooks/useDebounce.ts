import { useCallback, useRef } from 'react';

export const useDebounce = <TParam = unknown>(
  callback: (...args: TParam[]) => void,
  delay = 600,
) => {
  const refTime = useRef<NodeJS.Timeout | null>(null);

  const debouncedFn = useCallback(
    (...args: TParam[]) => {
      if (refTime.current) clearTimeout(refTime.current);
      refTime.current = setTimeout(() => {
        callback(...args);
      }, delay);
    },
    [callback, delay],
  );

  return debouncedFn;
};
