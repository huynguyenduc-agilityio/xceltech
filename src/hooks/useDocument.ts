import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';

// Constants
import { documentsQueryKeys, MESSAGES } from '@/constants';

// Types
import { MutationType } from '@/types';

// Services
import {
  downloadAllDocuments,
  getDocuments,
  addDocument,
  editDocument,
} from '@/services';

// Utils
import { downloadBlobFile } from '@/utils';

export const useGetDocuments = () => {
  const { data: documents, isFetching: isDocumentsLoading } = useQuery({
    queryKey: documentsQueryKeys.list(),
    queryFn: getDocuments,
  });

  return { documents, isDocumentsLoading };
};

export const useDocumentMutation = ({ type }: { type: MutationType }) => {
  const queryClient = useQueryClient();

  const mutationFn = type === MutationType.Create ? addDocument : editDocument;

  const {
    mutateAsync: handleDocumentMutation,
    isPending: isDocumentMutationLoading,
  } = useMutation({
    mutationFn,
    onSuccess: () => {
      queryClient.invalidateQueries({
        queryKey: documentsQueryKeys.list(),
      });
    },
  });

  return { handleDocumentMutation, isDocumentMutationLoading };
};

export const useDownloadDocuments = () => {
  const { isPending: isLoading, mutateAsync: handleDownloadDocuments } =
    useMutation({
      mutationFn: downloadAllDocuments,
      onSuccess: (data) => {
        if (!data) {
          throw new Error(MESSAGES.COMMON.DOWNLOAD_FAILED);
        }

        downloadBlobFile(data, 'documents.zip');
      },
    });

  return { isLoading, handleDownloadDocuments };
};
