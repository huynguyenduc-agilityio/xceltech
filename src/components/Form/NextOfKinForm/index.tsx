import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';

// Types
import {
  IEmployeeNextOfKinInfo,
  ToastStatus,
  NextOfKinFormValues,
} from '@/types';

// Components
import {
  Button,
  Form,
  FormControl,
  FormField,
  FormItem,
  FormMessage,
  Label,
  Select,
  Textarea,
  TextField,
} from '@/components';

// Constants
import { MESSAGES, RELATIONSHIP_OPTIONS } from '@/constants';

// Utils
import { nextOfKinSchema } from '@/utils/schemas/employee';

// Hooks
import { useToast, useUpdateInfoUser } from '@/hooks';

export interface INextOfKinForm {
  initialValues?: Partial<IEmployeeNextOfKinInfo>;
}

const NextOfKinForm = ({ initialValues }: INextOfKinForm) => {
  const { toast } = useToast();
  const { handleUpdateInfoUser, isUpdateInfoLoading } = useUpdateInfoUser();

  const {
    name = '',
    job = '',
    phone = '',
    residentialAddress = '',
    relationship = '',
  } = initialValues || {};
  const defaultValues: IEmployeeNextOfKinInfo = {
    name,
    job,
    phone,
    residentialAddress,
    relationship,
  };

  const form = useForm<NextOfKinFormValues>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    resolver: zodResolver(nextOfKinSchema),
    defaultValues,
  });

  const {
    control,
    handleSubmit,
    reset,
    getValues,
    formState: { isValid, isDirty, isSubmitting },
  } = form;

  const onSubmit = async (data: IEmployeeNextOfKinInfo) => {
    const payload = {
      kin: data,
    };

    try {
      await handleUpdateInfoUser(payload);

      toast({
        status: ToastStatus.Success,
        title: MESSAGES.COMMON.UPDATE_SUCCESS('Contact'),
      });
    } catch {
      toast({
        status: ToastStatus.Error,
        title: MESSAGES.COMMON.UPDATE_FAILED('Contact'),
      });
    } finally {
      reset(getValues());
    }
  };

  const disableSubmit = !isDirty || !isValid || isSubmitting;

  return (
    <Form {...form}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className="w-full px-10 py-16 text-black-default space-y-10">
          <div className="flex items-start gap-14">
            <FormField
              control={control}
              name="name"
              render={({ field, fieldState: { error } }) => (
                <TextField
                  label="Next of kin name"
                  variant="primary"
                  className="space-y-5"
                  labelClassName="text-md"
                  inputClassName="h-[68px] px-12 rounded-[15px]"
                  placeholder="Enter your next of kin name"
                  {...field}
                  errorMessage={error?.message}
                />
              )}
            />
            <FormField
              control={control}
              name="job"
              render={({ field, fieldState: { error } }) => (
                <TextField
                  label="Job / Occupation"
                  variant="primary"
                  className="space-y-5"
                  labelClassName="text-md"
                  inputClassName="h-[68px] px-12 rounded-[15px]"
                  placeholder="Enter your job / occupation"
                  {...field}
                  errorMessage={error?.message}
                />
              )}
            />
          </div>

          <div className="flex items-start gap-14">
            <FormField
              control={control}
              name="phone"
              render={({ field, fieldState: { error } }) => (
                <TextField
                  label="Phone Number"
                  variant="primary"
                  className="space-y-5"
                  labelClassName="text-md"
                  inputClassName="h-[68px] px-12 rounded-[15px]"
                  placeholder="Enter your phone number"
                  {...field}
                  errorMessage={error?.message}
                />
              )}
            />
            <FormField
              control={control}
              name="relationship"
              render={({ field }) => (
                <FormItem>
                  <Label className="text-md">Relationship</Label>
                  <FormControl>
                    <Select
                      option={RELATIONSHIP_OPTIONS}
                      placeholder="Select your relationship"
                      className="bg-blue-light h-[68px] px-12 rounded-[15px]"
                      {...field}
                      onChange={field.onChange}
                    />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />
          </div>

          <FormField
            control={control}
            name="residentialAddress"
            render={({ field }) => (
              <FormItem>
                <Label className="text-md">Residential Address</Label>
                <FormControl>
                  <Textarea
                    {...field}
                    className="h-[122px] px-9 rounded-[15px]"
                    placeholder="Enter your address"
                  />
                </FormControl>
                <FormMessage />
              </FormItem>
            )}
          />
          <div className="w-full h-[70px]">
            <Button
              type="submit"
              className="w-[364px] bg-green-primary"
              disabled={disableSubmit}
              isLoading={isUpdateInfoLoading}
            >
              Update
            </Button>
          </div>
        </div>
      </form>
    </Form>
  );
};

export default NextOfKinForm;
