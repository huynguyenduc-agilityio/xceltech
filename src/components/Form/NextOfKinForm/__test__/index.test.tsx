import userEvent from '@testing-library/user-event';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import NextOfKinForm from '..';

// Hooks
import { useToast, useUpdateInfoUser } from '@/hooks';

const mockHandleUpdateInfoUser = jest.fn();
const mockHandleUpdateInfoUserError = jest
  .fn()
  .mockRejectedValue(new Error('Network error'));
const mockToast = jest.fn();

jest.mock('@/hooks', () => ({
  useToast: jest.fn(),
  useUpdateInfoUser: jest.fn(),
}));

const renderComponent = () => {
  const queryClient = new QueryClient();

  return render(
    <QueryClientProvider client={queryClient}>
      <NextOfKinForm />
    </QueryClientProvider>,
  );
};

describe('NextOfKinForm Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (useToast as jest.Mock).mockReturnValue({ toast: mockToast });
    (useUpdateInfoUser as jest.Mock).mockReturnValue({
      isFamilyMutationLoading: false,
      handleUpdateInfoUser: mockHandleUpdateInfoUser,
    });
  });

  it('should match snapshot', () => {
    const container = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('handle input value change and form submit', async () => {
    const { getByPlaceholderText, container, getByRole } = renderComponent();

    const nameInput = getByPlaceholderText('Enter your next of kin name');
    const jobInput = getByPlaceholderText('Enter your job / occupation');
    const phoneInput = getByPlaceholderText('Enter your phone number');
    const addressInput = getByPlaceholderText('Enter your address');
    const relationshipSelect = container.querySelector(
      'select[name="relationship"]',
    ) as HTMLSelectElement;

    const submitButton = getByRole('button', { name: 'Update' });

    fireEvent.change(nameInput, { target: { value: 'John Doe' } });
    fireEvent.change(jobInput, { target: { value: 'Software Engineer' } });
    fireEvent.change(phoneInput, { target: { value: '0987654321' } });
    fireEvent.change(addressInput, { target: { value: '123 Main Street' } });

    fireEvent.change(relationshipSelect, {
      target: { value: 'relative' },
    });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockHandleUpdateInfoUser).toHaveBeenCalled();
    });
  });

  it('handles form submit error gracefully', async () => {
    (useUpdateInfoUser as jest.Mock).mockReturnValue({
      isLoading: false,
      handleUpdateInfoUser: mockHandleUpdateInfoUserError,
    });

    const { getByPlaceholderText, container, getByRole } = renderComponent();

    const nameInput = getByPlaceholderText('Enter your next of kin name');
    const jobInput = getByPlaceholderText('Enter your job / occupation');
    const phoneInput = getByPlaceholderText('Enter your phone number');
    const addressInput = getByPlaceholderText('Enter your address');
    const relationshipSelect = container.querySelector(
      'select[name="relationship"]',
    ) as HTMLSelectElement;

    const submitButton = getByRole('button', { name: 'Update' });

    fireEvent.change(nameInput, { target: { value: 'John Doe' } });
    fireEvent.change(jobInput, { target: { value: 'Software Engineer' } });
    fireEvent.change(phoneInput, { target: { value: '0987654321' } });
    fireEvent.change(addressInput, { target: { value: '123 Main Street' } });

    fireEvent.mouseDown(relationshipSelect);

    fireEvent.change(relationshipSelect, {
      target: { value: 'relative' },
    });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockToast).toHaveBeenCalled();
    });
  });

  it('should show error message fields', async () => {
    const { getByPlaceholderText, getByText } = renderComponent();

    const nameInput = getByPlaceholderText('Enter your next of kin name');
    const jobInput = getByPlaceholderText('Enter your job / occupation');
    const phoneInput = getByPlaceholderText('Enter your phone number');
    const addressInput = getByPlaceholderText('Enter your address');

    fireEvent.change(nameInput, { target: { value: '' } });
    fireEvent.change(jobInput, { target: { value: '' } });
    fireEvent.change(phoneInput, { target: { value: '' } });
    fireEvent.change(addressInput, { target: { value: '' } });

    fireEvent.blur(nameInput);
    fireEvent.blur(jobInput);
    fireEvent.blur(phoneInput);
    fireEvent.blur(addressInput);

    await waitFor(() => {
      expect(getByText('Next of kin name is required!')).toBeInTheDocument();
      expect(getByText('Job / Occupation is required!')).toBeInTheDocument();
      expect(getByText('Phone Number is required!')).toBeInTheDocument();
      expect(getByText('Residential Address is required!')).toBeInTheDocument();
    });
  });
});
