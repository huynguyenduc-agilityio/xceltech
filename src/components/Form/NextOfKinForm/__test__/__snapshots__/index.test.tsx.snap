// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextOfKinForm Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="w-full px-10 py-16 text-black-default space-y-10"
        >
          <div
            class="flex items-start gap-14"
          >
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Next of kin name-label"
                >
                  Next of kin name
                </label>
                <input
                  aria-describedby=":r0:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Next of kin name-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r0:-form-item"
                  name="name"
                  placeholder="Enter your next of kin name"
                  value=""
                />
              </div>
            </div>
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Job / Occupation-label"
                >
                  Job / Occupation
                </label>
                <input
                  aria-describedby=":r1:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Job / Occupation-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r1:-form-item"
                  name="job"
                  placeholder="Enter your job / occupation"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="flex items-start gap-14"
          >
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Phone Number-label"
                >
                  Phone Number
                </label>
                <input
                  aria-describedby=":r2:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Phone Number-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r2:-form-item"
                  name="phone"
                  placeholder="Enter your phone number"
                  value=""
                />
              </div>
            </div>
            <div
              class="space-y-2 w-full"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
              >
                Relationship
              </label>
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r4:"
                aria-expanded="false"
                class="font-productsans flex w-full items-center justify-between border border-input py-2 text-lg ring-offset-background data-[placeholder]:text-gray-soft focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-placeholder=""
                data-state="closed"
                dir="ltr"
                role="combobox"
                title="Select"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  Select your relationship
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
              <select
                aria-hidden="true"
                name="relationship"
                style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
                tabindex="-1"
              >
                <option
                  value="relative"
                >
                  Relative
                </option>
                <option
                  value="independent"
                >
                  Independent
                </option>
              </select>
            </div>
          </div>
          <div
            class="space-y-2 w-full"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
            >
              Residential Address
            </label>
            <textarea
              aria-describedby=":r5:-form-item-description"
              aria-invalid="false"
              class="flex min-h-[80px] w-full rounded-regular bg-blue-light py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none resize-none disabled:cursor-not-allowed disabled:opacity-50 h-[122px] px-9 rounded-[15px]"
              id=":r5:-form-item"
              name="residentialAddress"
              placeholder="Enter your address"
            />
          </div>
          <div
            class="w-full h-[70px]"
          >
            <button
              class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[364px] bg-green-primary"
              disabled=""
              type="submit"
            >
              Update
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="w-full px-10 py-16 text-black-default space-y-10"
      >
        <div
          class="flex items-start gap-14"
        >
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Next of kin name-label"
              >
                Next of kin name
              </label>
              <input
                aria-describedby=":r0:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Next of kin name-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r0:-form-item"
                name="name"
                placeholder="Enter your next of kin name"
                value=""
              />
            </div>
          </div>
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Job / Occupation-label"
              >
                Job / Occupation
              </label>
              <input
                aria-describedby=":r1:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Job / Occupation-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r1:-form-item"
                name="job"
                placeholder="Enter your job / occupation"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="flex items-start gap-14"
        >
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Phone Number-label"
              >
                Phone Number
              </label>
              <input
                aria-describedby=":r2:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Phone Number-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r2:-form-item"
                name="phone"
                placeholder="Enter your phone number"
                value=""
              />
            </div>
          </div>
          <div
            class="space-y-2 w-full"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
            >
              Relationship
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r4:"
              aria-expanded="false"
              class="font-productsans flex w-full items-center justify-between border border-input py-2 text-lg ring-offset-background data-[placeholder]:text-gray-soft focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-blue-light h-[68px] px-12 rounded-[15px]"
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              role="combobox"
              title="Select"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Select your relationship
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              name="relationship"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                value="relative"
              >
                Relative
              </option>
              <option
                value="independent"
              >
                Independent
              </option>
            </select>
          </div>
        </div>
        <div
          class="space-y-2 w-full"
        >
          <label
            class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
          >
            Residential Address
          </label>
          <textarea
            aria-describedby=":r5:-form-item-description"
            aria-invalid="false"
            class="flex min-h-[80px] w-full rounded-regular bg-blue-light py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none resize-none disabled:cursor-not-allowed disabled:opacity-50 h-[122px] px-9 rounded-[15px]"
            id=":r5:-form-item"
            name="residentialAddress"
            placeholder="Enter your address"
          />
        </div>
        <div
          class="w-full h-[70px]"
        >
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[364px] bg-green-primary"
            disabled=""
            type="submit"
          >
            Update
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
